src/components/Paginate.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { FunctionComponent, useEffect, useState } from "react";
+
+const getPageCount = (
+  numberOfResults: number,
+  numberOfResultsPerPage: number
+): number => Math.ceil(numberOfResults / numberOfResultsPerPage);
+
+interface PaginateProps {
+  children: FunctionComponent;
+  limit: number;
+  onPageClick: (page: number) => void;
+  page: number;
+  totalResults: number;
+}
+
+export default function Paginate({
+  children,
+  limit,
+  onPageClick,
+  page,
+  totalResults,
+}: PaginateProps) {
+  const [currentPage, setCurrentPage] = useState(page);
+  const numberOfPages: number = getPageCount(totalResults, limit);
+
+  useEffect(() => {
+    onPageClick(currentPage);
+  }, [currentPage]);
```

Comment:
Because `currentPage` is in `useEffect`'s dependency array, `useEffect` will not fire if, say, users click on page 2 while they are already on page 2.

This is important for two reasons: first, and perhaps more importantly, the page will not refresh if users select the current page, saving some bandwidth, and more importantly, time; second, we save an API request, since we won't have to ask for the same page of search results again if we haven't changed pages.

@akshara-sun I would make sure you're aware of `useEffect`'s dependency array and how it saves re-renders.